### PR TITLE
WIP: Build for py38

### PR DESCRIPTION
**Pull request by stefanv**
This probably depends on having wheels available for NumPy etc. (they aren't, yet).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -49,6 +49,17 @@ matrix:
...
+      - NP_TEST_DEP=numpy==1.14.5
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.8
+        - NP_BUILD_DEP=numpy==1.14.5
+        - NP_TEST_DEP=numpy==1.14.5
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.8
+        - PLAT=i686
```
Comment (hmaarrfk) [inline .travis.yml]: 32 bit is still supported on OSX?

File: .appveyor.yml [hunk 2]
```diff
@@ -27,6 +27,12 @@ environment:
     - PYTHON: C:\Python37-x64
       NP_BUILD_DEP: "numpy==1.14.5"
```
Comment (hmaarrfk) [inline .appveyor.yml]: Numpy build time dependency got bumped to 0.14 for all builds

https://github.com/scikit-image/scikit-image/blob/master/requirements/build.txt#L3

File: .travis.yml (hunk 1 above)
Comment (stefanv) [reply .travis.yml]: This is for Linux.

File: .appveyor.yml (hunk 2 above)
Comment (stefanv) [reply .appveyor.yml]: So this is correct, then, right?

File: .travis.yml (hunk 1 above)
Comment (hmaarrfk) [reply .travis.yml]: For some reason II thought MB = MacBook

File: .appveyor.yml (hunk 2 above)
Comment (hmaarrfk) [reply .appveyor.yml]: The ones above it are set to 0.13 i think.

Comment (stefanv) [reply .appveyor.yml]: Gotcha, fixed.

**Comment by jni** [comment]
@stefanv please remember to keep us posted with regards to NumPy and SciPy 3.8 wheels!

File: .travis.yml [hunk 3]
```diff
@@ -31,37 +31,36 @@ matrix:
...
-        - NP_TEST_DEP=numpy==1.14.5
     - os: linux
       env:
         - MB_PYTHON_VERSION=3.7
         - PLAT=i686
-        - NP_BUILD_DEP=numpy==1.14.5
-        - NP_TEST_DEP=numpy==1.14.5
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.8
```
Comment (hmaarrfk) [inline .travis.yml]:
```suggestion
        - MB_PYTHON_VERSION=3.8
        - NP_BUILD_DEP=numpy==1.17.3
        - NP_TEST_DEP=numpy==1.17.3
```

File: .travis.yml [hunk 4]
```diff
@@ -31,37 +31,36 @@ matrix:
...
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.8
```
Comment (hmaarrfk) [inline .travis.yml]:
```suggestion
        - MB_PYTHON_VERSION=3.8
        - NP_BUILD_DEP=numpy==1.17.3
        - NP_TEST_DEP=numpy==1.17.3
```

File: .travis.yml [hunk 5]
```diff
@@ -31,37 +31,36 @@ matrix:
...
     - os: osx
       language: generic
       env:
         - MB_PYTHON_VERSION=3.7
-        - NP_BUILD_DEP=numpy==1.14.5
-        - NP_TEST_DEP=numpy==1.14.5
+    - os: osx
+      language: generic
+      env:
+        - MB_PYTHON_VERSION=3.8
```
Comment (hmaarrfk) [inline .travis.yml]:
```suggestion
        - MB_PYTHON_VERSION=3.8
        - NP_BUILD_DEP=numpy==1.17.3
        - NP_TEST_DEP=numpy==1.17.3
```

File: .appveyor.yml [hunk 6]
```diff
@@ -27,6 +27,12 @@ environment:
...
       NP_TEST_DEP: "numpy==1.14.5"
+    - PYTHON: C:\Python38
+      NP_BUILD_DEP: "numpy==1.14.5"
```
Comment (hmaarrfk) [inline .appveyor.yml]:
```suggestion
      NP_BUILD_DEP: "numpy==1.17.3"
```

File: .appveyor.yml [hunk 7]
```diff
@@ -27,6 +27,12 @@ environment:
...
+      NP_TEST_DEP: "numpy==1.14.5"
```
Comment (hmaarrfk) [inline .appveyor.yml]:
```suggestion
      NP_TEST_DEP: "numpy==1.17.3"
```

File: .appveyor.yml [hunk 8]
```diff
@@ -27,6 +27,12 @@ environment:
...
+    - PYTHON: C:\Python38-x64
+      NP_BUILD_DEP: "numpy==1.14.5"
```
Comment (hmaarrfk) [inline .appveyor.yml]:
```suggestion
      NP_BUILD_DEP: "numpy==1.17.3"
```

File: .appveyor.yml [hunk 9]
```diff
@@ -27,6 +27,12 @@ environment:
...
+      NP_TEST_DEP: "numpy==1.14.5"
```
Comment (hmaarrfk) [inline .appveyor.yml]:
```suggestion
      NP_TEST_DEP: "numpy==1.17.3"
```

**Comment by hmaarrfk** [comment]
@stefanv numpy only built 1.17.3 for python 3.8

**Comment by stefanv** [comment]
@mattip Any reason why we can't build some older NumPy wheels for 3.8?  I suspect not?

**Comment by hmaarrfk** [comment]
@stefanv whats the point of building older numpy for newer versions of python? I don't think it is too good to do that, and we can simply build with the newer versions.

**Comment by stefanv** [comment]
@hmaarrfk There are probably a lot of `==` dependencies on old versions of NumPy out there, so it would make sense to make older versions available. But, for us, you're right: we can simply start relying on newer versions for our 3.8 builds.

**Comment by hmaarrfk** [comment]
This can now be closed.

Thanks @stefanv for early efforts here!

**Pull request closed without merging**